src/keys/get-management-token.ts

```diff
@@ -67,25 +70,45 @@ const getTokenFromOneTimeToken = async (
     `Successfully retrieved CMA Token for app ${appInstallationId} in space ${spaceId} and environment ${environmentId}`
   )
 
-  return JSON.parse(response.body).token
+  return JSON.parse(response.body).token as Promise<string>
 }
 
 /**
  * Factory method for GetManagementToken
  * @internal
  */
 export const createGetManagementToken = (log: Logger, http: HttpClient) => {
```

Comment:
I would inject `cache` here, so we can test different behaviors with a mocked cache (for example different `ttl`s)

Reply:
I thought about that, but since the TTL comes from the token and caching can be disabled over the `reuseToken` boolean, there wasn't too much use for that. I will still do it because there might be some tests in the future that use it.